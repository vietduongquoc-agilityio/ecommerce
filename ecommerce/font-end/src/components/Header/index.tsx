"use client";

import { Box, Flex } from "@radix-ui/themes";
import Image from "next/image";
import logo from "../../../public/Images/logo.png";
import cart from "../../../public/Images/cart.png";
import NavLink from "../NavLink";
import styles from "./index.module.css";

export default function Header() {
  return (
    <Box width="100%" maxWidth="1286px">
      <Flex align="center" justify="between">
        <Box>
          <Image src={logo} alt="logo" className={`${styles.image} ${styles.logo}`} />
        </Box>
        <Flex align="center" justify="between" width="430px">
          <NavLink href="">Home</NavLink>
          <NavLink href="">Shop</NavLink>
          <NavLink href="">About</NavLink>
          <NavLink href="">Contact</NavLink>
        </Flex>
        <Image src={cart} width="25" height="22" alt="cart-icon" className={`${styles.image} ${styles.cart}`} />
      </Flex>
    </Box>
  );
}
