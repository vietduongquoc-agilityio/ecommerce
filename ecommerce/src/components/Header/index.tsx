"use client";

import { Box, Flex } from "@radix-ui/themes";
import Image from "next/image";
import logo from "@/assets/Images/Logo.png";
import styles from "./style.module.css";
import Link from "next/link";
import cart from "@/assets/Images/Cart.png"

export default function Header() {
  return (
    <Box width="100%" maxWidth="1286px">
      <Flex align="center" justify="between">
        <Box>
          <Image src={logo} alt="logo" className={`${styles.image} ${styles.logo}`} />
        </Box>
        <Flex align="center" justify="between" width="430px">
          <Link href="" className={styles.link}>Home</Link>
          <Link href="" className={styles.link}>Shop</Link>
          <Link href="" className={styles.link}>About</Link>
          <Link href="" className={styles.link}>Contact</Link>
        </Flex>
        <Image src={cart} width="25" height="22" alt="cart-icon" className={`${styles.image} ${styles.cart}`} />
      </Flex>
    </Box>
  );
}
