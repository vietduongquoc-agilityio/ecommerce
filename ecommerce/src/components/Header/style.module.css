.image {
  transition: transform 0.3s ease;
}

.image:hover {
  transform: scale(1.1);
}

.logo {
  width: 170px;
  cursor: pointer;
}

.cart {
  width: 25px;
  cursor: pointer;
}

.link {
    font-size: 16px;
    font-weight: 500;
    padding: 0 10px;
  }
  
  .link:hover {
    color: orange;
  }
  