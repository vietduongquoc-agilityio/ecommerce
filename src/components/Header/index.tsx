"use client";

import { Box, Flex } from "@radix-ui/themes";
import Link from "next/link";
import Image from "next/image";
import { useRouter } from "next/navigation";
import { colors, fonts } from "@/themes";

// Images
import logo from "@/assets/Images/Logo.png";

// Components
import { CartPopover } from "..";

const Header = () => {
  const router = useRouter();

  const menuItems = [
    { name: "Home", path: "/" },
    { name: "Shop", path: "/shop" },
    { name: "About", path: "/about" },
    { name: "Contact", path: "/contact" },
  ];

  return (
    <Box
      width="100%"
      maxWidth="1286px"
      style={{
        height:"100px",
        margin: "0 auto",
      }}
    >
      <section
        style={{
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center",
          height:"100%"
        }}
      >
        {/* Logo */}
        <Box>
          <Image
            src={logo}
            alt="logo"
            style={{
              width: "170px",
              cursor: "pointer",
              transition: "transform 0.3s ease",
            }}
            onMouseOver={(e) =>
              (e.currentTarget.style.transform = "scale(1.1)")
            }
            onMouseOut={(e) => (e.currentTarget.style.transform = "scale(1)")}
            onClick={() => router.push("/")}
          />
        </Box>

        {/* Navigation Links */}
        <Flex align="center" justify="between" width="430px">
          {menuItems.map((item) => (
            <Link
              key={item.name}
              href={item.path}
              onClick={() => router.push(item.path)}
              style={{
                fontSize: fonts.size.xs,
                fontWeight: fonts.weight.medium,
                padding: "0 10px",
                textDecoration: "none",
                color: colors.secondary.black,
                transition: "color 0.3s ease",
                cursor: "pointer",
              }}
              onMouseOver={(e) =>
                (e.currentTarget.style.color = colors.primary.yellow)
              }
              onMouseOut={(e) =>
                (e.currentTarget.style.color = colors.secondary.black)
              }
            >
              {item.name}
            </Link>
          ))}
        </Flex>

        <CartPopover />
      </section>
    </Box>
  );
};

export default Header;
