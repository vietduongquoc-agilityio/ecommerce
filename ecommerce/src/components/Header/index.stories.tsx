import type { Meta, StoryObj } from "@storybook/react";
import Header from "./";
import { Theme } from "@radix-ui/themes";

const meta: Meta<typeof Header> = {
  title: "Components/Header",
  component: Header,
  decorators: [
    (Story) => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof Header>;

export const Default: Story = {};
