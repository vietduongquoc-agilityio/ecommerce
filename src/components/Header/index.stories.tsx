import type { Meta, StoryObj } from "@storybook/react";
import Header from "./";

const meta: Meta<typeof Header> = {
  title: "Components/Header",
  component: Header,
  tags: ['autodocs'],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;

type Story = StoryObj<typeof Header>;

export const Default: Story = {};