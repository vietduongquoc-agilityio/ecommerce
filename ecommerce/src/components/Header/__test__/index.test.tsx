import { render } from "@testing-library/react";
import Header from "../Header";

describe("Header Component", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
