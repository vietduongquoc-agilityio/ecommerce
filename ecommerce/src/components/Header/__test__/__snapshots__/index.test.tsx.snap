// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="radix-themes"
    data-accent-color="indigo"
    data-gray-color="slate"
    data-has-background="true"
    data-is-root-theme="true"
    data-panel-background="translucent"
    data-radius="medium"
    data-scaling="100%"
  >
    <div
      class="rt-Box rt-r-w rt-r-max-w"
      style="--width: 100%; --max-width: 1286px;"
    >
      <div
        class="rt-Flex rt-r-ai-center rt-r-jc-space-between"
      >
        <div
          class="rt-Box"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent; width: 170px; cursor: pointer; transition: transform 0.3s ease;"
            width="40"
          />
        </div>
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-space-between rt-r-w"
          style="--width: 430px;"
        >
          <a
            href=""
            style="font-size: 16px; font-weight: 500; padding: 0px 10px; text-decoration: none; color: rgb(58, 58, 58); transition: color 0.3s ease;"
          >
            Home
          </a>
          <a
            href=""
            style="font-size: 16px; font-weight: 500; padding: 0px 10px; text-decoration: none; color: rgb(58, 58, 58); transition: color 0.3s ease;"
          >
            Shop
          </a>
          <a
            href=""
            style="font-size: 16px; font-weight: 500; padding: 0px 10px; text-decoration: none; color: rgb(58, 58, 58); transition: color 0.3s ease;"
          >
            About
          </a>
          <a
            href=""
            style="font-size: 16px; font-weight: 500; padding: 0px 10px; text-decoration: none; color: rgb(58, 58, 58); transition: color 0.3s ease;"
          >
            Contact
          </a>
        </div>
        <img
          alt="cart-icon"
          data-nimg="1"
          decoding="async"
          height="22"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
          style="color: transparent; cursor: pointer; transition: transform 0.3s ease;"
          width="25"
        />
      </div>
    </div>
  </div>
</div>
`;
